bonus_guides/X_seeding_data.md

```diff
@@ -0,0 +1,53 @@
+When creating an app, it's important that we're able to seed our database with some initial data (e.g., for early development work or pre-launch testing purposes).
+
+Fortunately, Phoenix already provides us with a convention for seeding data. By default Phoenix generates a script file for each app at `priv/repo/seeds.exs`, which we can use to use to populate our database/repository.
+
+Also note that in order to seed data as in the example below you should have alrady generated and run the related migration (i.e., Link migration, controller, model, etc.) and updated your `router.ex`, as described in the [Ecto Models Guide](http://www.phoenixframework.org/docs/ecto-models) (if you haven't completed that Guide yet, should should do so before proceeding further).
+
+So in order to seed data, we simply need to add a script to `seeds.exs` that uses our database/repository to directly add the data we want. As you can see from the comments that Phoenix generated for us in `seeds.exs` file, we should follow this pattern:
+
+```elixir
+  <%= application_module %>.Repo.insert!(%<%= application_module %>.SomeModel{})
+```
+For example, if we were creating an app called Linker and wanted to seed a Link table in our database/repository with with a series of links, we could simply add the following script to our `seeds.exs` file:
+
+```elixir
+  ...
+  alias Linker.Repo
+  alias Linker.Link
+
+  Repo.insert! %Link{
+    title: "Phoenix Framework",
+    url: "http://www.phoenixframework.org/"
+  }
+  
+  Repo.insert! %Link{
+    title: "Elixir",
+    url: "http://elixir-lang.org/"
+  }
+  
+  Repo.insert! %Link{
+    title: "Erlang",
+    url: "https://www.erlang.org/"
+  }
+  ...
+```
+With this script, we've set up some aliases and then progressed through a list of our Links which will each be written to the database/repo when we run the `seeds.exs` file with mix run:
+
+```elixir
+  mix run priv/repo/seeds.exs
+```
+Note that if we wanted to delete/scrub all prior data that we seeded in the Link table, we could also include `Repo.delete_all Link` in your script immediate above `Repo.insert!` 
+
+#### Models are Initialized
+
+Conveniently, when following this convention, Phoenix makes sure that our models are appropriately initialized; and as long as we use the bang functions (e.g.,  `insert!`, `update!`, etc.), they will also fail if something goes wrong.
```

Comment:
I think we can turn the semicolon into a period. Then we can get rid of the "and" and capitalize the "a" in "as". Let me be more clear:

. . . are appropriately initialized. As long as . . .